x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_significant_events_view/index.tsx

```diff
@@ -244,14 +247,46 @@ export function StreamDetailSignificantEventsView({ definition }: Props) {
           </EuiFlexGroup>
         </EuiFlexItem>
 
-        <EuiFlexItem grow={false}>
-          <PreviewDataSparkPlot
-            definition={definition.stream}
-            query={{ kql: { query: combinedQuery ?? '' }, id: 'preview_all', title: 'All events' }}
-            isQueryValid={true}
-            noOfBuckets={50}
-          />
-        </EuiFlexItem>
+        <EuiPanel grow={false} hasShadow={false} hasBorder={true}>
+          <EuiFlexGroup direction="column" gutterSize="xs">
+            <EuiFlexItem grow={false}>
+              <EuiText css={{ fontWeight: euiTheme.font.weight.semiBold }}>
+                {i18n.translate(
+                  'xpack.streams.addSignificantEventFlyout.manualFlow.previewChartDetectedOccurrences',
+                  {
+                    defaultMessage: 'Detected event occurrences ({count})',
+                    values: {
+                      count: (significantEventsFetchState.value?.all ?? []).reduce(
+                        (acc, point) => acc + point.y,
+                        0
+                      ),
+                    },
+                  }
+                )}
+              </EuiText>
+            </EuiFlexItem>
+
+            <EuiFlexItem grow={false}>
+              <SignificantEventsHistogramChart
+                id={'all-events'}
+                occurrences={significantEventsFetchState.value?.all ?? []}
```

Comment:
It feels weird to me that the histogram reacts to the time filter but not to the query.

Are we sure this is the expected behavior?

Reply:
Yes!

Reply:
I moved the filtering to the server in https://github.com/elastic/kibana/pull/246046/commits/78e4a83fed3e70bf07d02686b704b3d8a61f3dbe